process-engine/bpmn-studio
# 🚧 WIP Feature/property panel

## What did you change?

This branch adds the new Property Panel, this includes:

1) general-indextab 

The general indextabs duty is to managa the basic information about most of the elements.

Namely: 
- basics-section (id, name, documentation)
- call-activity-section (called element)
- error-event (error-name, code and message)
- message-event (message-id and name)
- escalation-event (escalation-id, name, code, code-variable)
- flow (condition [only if gateway is there])
- pool (version-tag, process-id, process-name)
- script-task (script-format, script, result variable)
- signal-event (signal-id, name)

2) forms-tab

This indextab offers the configuration possibilities that are necessary to create forms on a user-task.

Namely:
- basics-section (form-key, form-field, id, type, label, default)

3) extensions-tab

This tab mimics the behaviour of the camunda extensions-tab.

This branch also includes a basic color picker for coloring your diagrams even more awesomely.

## How can others test the changes?

- Checkout the branch.
- Start bpmn-studio.
- Create a new process and open it via the "details"-button.
- See and test the new property panel on the right-hand side.

## Related Issues

Closes #130 

## PR-Checklist

Please check the boxes in this list after submitting your PR:
